### DISCUSSION TITLE
Bot cant close currently opened window

### DISCUSSION THREAD

**Discussion by pwpretz** [PrismarineJS/mineflayer#3152] [Q&A] [answered]
my code is 

function sell() {
  bot.chat("/ ds qsell")
  bot.on("windowOpen", (window) => {
    bot.inventory.slots.forEach((item) => {
      if (item) {
        if (item.type === 859) {
          window.deposit(859, null, 64)
        }
      }
    })
    bot.closeWindow(window)
    console.log("sold")
  })
}

ive tried replacing window with current window like this:

bot.closeWindow(bot.currentWindow)

**Comment by amoraschi** [comment]
Read the issue template, provide logs, expected behavior, current behavior, etc.

**Reply by pwpretz** [reply]
it doesnt give error
it just get stucks and then crashes due to ram 

.
i want it to close the window it doesnt

**Comment by pwpretz** [comment, marked as answer]
i just put the close window function outside of bot.on('windowOpen")
